react/w-books/src/app/components/Routes/index.tsx

```diff
@@ -8,7 +10,11 @@ function Routes() {
       <BrowserRouter>
         <Switch>
           {
-            ROUTES.map(route => <Route key={route.path} component={route.component} path={route.path} exact={route.exact} />)
+            ROUTES.map(route => 
+              route.private ? 
```

Comment:
with de router.private, you could send by prop to the new component and inside it validate the conditions